tools/js-sdk-release-tools/src/changelog/v2/ChangelogGenerator.ts

```diff
@@ -0,0 +1,690 @@
+  private enumMemberRemovedTemplate = 'Enum {enumName} no longer has value {valueName}';
+
+  private changelogItems: ChangelogItems = {
+    features: new Map<ChangelogItemCategory, string[]>(),
+    breakingChanges: new Map<ChangelogItemCategory, string[]>(),
+  };
+
+  constructor(
+    private detectContext: DetectContext,
+    private detectResult: DetectResult
+  ) {}
+
+  // TODO: add enum support
+  public generate(): ChangelogResult {
+    this.detectResult.interfaces.forEach((diffPairs, name) => {
+      this.generateForInterfaces(diffPairs, name);
+    });
+    this.detectResult.classes.forEach((diffPairs, name) => {
+      this.generateForClasses(diffPairs, name);
+    });
+    this.detectResult.typeAliases.forEach((diffPairs, name) => {
+      this.generateForTypeAliases(diffPairs, name);
+    });
+    this.detectResult.functions.forEach((diffPairs, name) => {
+      this.generateForFunctions(diffPairs, name);
+    });
+    this.detectResult.enums.forEach((diffPairs, name) => {
+      this.generateForEnums(diffPairs, name);
+    });
+    const content = this.generateContentCore();
+    const hasBreakingChange = this.hasBreakingChange();
+    const hasFeature = this.hasFeature();
+    const changelogItems = this.getChangelogItems();
+    const breakingChangeItems = Array.from(this.changelogItems.breakingChanges).flatMap(([_, items]) => items);
+    return { content, changelogItems, hasBreakingChange, hasFeature, breakingChangeItems };
+  }
+
+  private hasBreakingChange(): boolean {
+    return this.changelogItems.breakingChanges.size > 0;
+  }
+
+  private hasFeature(): boolean {
+    return this.changelogItems.features.size > 0;
+  }
+
+  private getChangelogItems(): ChangelogItems {
+    return this.changelogItems;
+  }
+
+  private getItemsFromCategoryMap(map: Map<ChangelogItemCategory, string[]>): string[] {
+    const items: string[] = [];
+    [...map.keys()].sort().forEach((category) => {
+      const categoryItems = map.get(category);
+      if (!categoryItems || categoryItems.length === 0) return;
+      categoryItems.forEach((item) => items.push(item));
+    });
+    return items;
+  }
+
+  private generateSection(items: string[], title: string): string {
+    if (items.length === 0) return '';
+    let content = `### ${title}\n`;
+    content += items.map((i) => `  - ${i}`).join('\n');
+    content += '\n';
+    return content;
+  }
+
+  private generateContentCore() {
+    let content = '';
+    const featureItems = this.getItemsFromCategoryMap(this.changelogItems.features);
+    content += this.generateSection(featureItems, 'Features Added');
+
+    const breakingChangeItems = this.getItemsFromCategoryMap(this.changelogItems.breakingChanges);
+    content += this.generateSection(breakingChangeItems, 'Breaking Changes');
+
+    return content;
+  }
+
+  private addChangelogItem(category: ChangelogItemCategory, message: string) {
+    if (category < 1000) {
```

Comment:
Can we have a virtual enum member draw a line between non-breaking and breaking and change this place into that enum member instead?